test/org/sagebionetworks/bridge/services/HealthDataServiceTest.java

```diff
@@ -123,4 +140,43 @@ public void getRecordsForUploadDateSuccess() {
         assertEquals("bar healthcode", recordList.get(1).getHealthCode());
         assertEquals("baz healthcode", recordList.get(2).getHealthCode());
     }
+
+    @Test
+    public void updateRecordSuccess() throws Exception {
+
+        // first create a mock record
+        // record
+        HealthDataRecord record = DAO.getRecordBuilder().withHealthCode(TEST_HEALTH_CODE)
+                .withSchemaId(TEST_SCHEMA_ID).withSchemaRevision(3).withStudyId(TEST_STUDY_ID).build();
+
+        // mock dao
+        HealthDataDao mockDao = mock(HealthDataDao.class);
+        when(mockDao.createOrUpdateRecord(record)).thenReturn(TEST_RECORD_ID);
+
+        HealthDataService svc = new HealthDataService();
+        svc.setHealthDataDao(mockDao);
+
+        // execute and validate
+        String retVal = svc.createOrUpdateRecord(record);
+        assertEquals(TEST_RECORD_ID, retVal);
+
+        // then create a mock json request
+        RecordExportStatusRequest recordExportStatusRequest = createMockRecordExportStatusRequest();
+        when(mockDao.getRecordById(TEST_RECORD_ID)).thenReturn(record);
+
+        // finally call service method and assert
+        svc.updateRecordsWithExporterStatus(recordExportStatusRequest);
+        HealthDataRecord recordAfter = svc.getRecordById(TEST_RECORD_ID);
+        assertNotNull(recordAfter.getSynapseExporterStatus());
+        assertEquals(HealthDataRecord.ExporterStatus.SUCCEEDED, recordAfter.getSynapseExporterStatus());
+
+    }
+
+    private RecordExportStatusRequest createMockRecordExportStatusRequest() throws Exception {
+        RecordExportStatusRequest request = new RecordExportStatusRequest();
+        request.setRecordIds(Arrays.asList(TEST_RECORD_ID));
```

Comment:
Might consider putting two records in here just to make sure any iteration that needs to occur is occurring (true for the controller test as well).